consensus/state.go

```diff
@@ -1117,18 +1117,23 @@ func (cs *State) defaultDecideProposal(height int64, round int32) {
 		cs.Logger.Error("enterPropose: Error signing proposal", "height", height, "round", round, "err", err)
 	}
 
+	// TODO(ismail): capture this in the Consensus ADR
 	// post data to ipfs
 	// TODO(evan): don't hard code context and timeout
-	if cs.ipfs != nil {
```

Comment:
Because this is no longer optional, I had to change a lot of (consensus reactor) tests to set the ipfsAPI. If that is too difficult to review, I can split that change out into a separate PR.

Reply:
This resulted in a bunch of failing tests as every mock ipfs object is not really lightweight (many go-routines). 
As a result I changed:
- https://github.com/lazyledger/lazyledger-core/pull/323/commits/a295da45bfeffd2e79438a920403be58ee348491
- https://github.com/lazyledger/lazyledger-core/pull/323/commits/79457ae8b30fa504b586c2c995c06b9a8d31cb7d
- https://github.com/lazyledger/lazyledger-core/pull/323/commits/c867b7826aae11607013093070c11c6f504bd72e